openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -125,6 +127,11 @@ def process_reference(self, context, name, namespace, options):
 
             return new_nodes
 
+    def load(self, context, name=None, namespace=None, options=None):
+        super(ReferenceLoader, self).load(context, name, namespace, options)
+        # clean containers if present to AVALON_CONTAINERS
+        self._organize_containers(self[:])
```

Comment:
Another question - what's the reason we want this? Isn't it much nicer if they actually remain within the container of what loaded it? You'd get these containers in the containers

E.g. I think this is much better:
```
AVALON_CONTAINERS > layoutMain > modelMain
```
Than:
```
AVALON_CONTAINERS > layoutMain
AVALON_CONTAINERS > modelMain
```

Reply:
> Without any organizing my loaded containers loads to the scene root and because they are not hidden in the Outliner, they start to clutter it

Ah, of course. With the change where only reference nodes are added into the avalon container then of course other scene nodes aren't added in there either. I guess that's a good thing.

Then one more question. What's the reason this organizing isn't added directly to the `openpype.hosts.maya.api.plugin.ReferenceLoader` class but only on the inherited one?

Reply:
> Then one more question. What's the reason this organizing isn't added directly to the `openpype.hosts.maya.api.plugin.ReferenceLoader` class but only on the inherited one? I'd recommend moving it there.

Note that this is also still an open question. It'd reduce the amount of lines of code.

Reply:
I've moved the code to the parent.